### PR TITLE
Fix conan_export.tgz error by adding missing download route.

### PR DESCRIPTION
**Pull request by remgodow**
Fix for issue #36 

This pull request makes the following changes:
* adds route for downloading conan_export.tgz
* adds separate AssetKind for conan_export.tgz (probably it is not required, but I changed it for code readability)

It can be tested on its own via NEXUS Browse page, testing via Conan client requires fix for download_url #34, which has not been merged to master.

### CODE REVIEW

**Comment by doddi** [comment]
@remgodow nice work and thanks for the contribution! If you could sign the cla and follow the instructions above I will get it merged in.

Again thanks, we appreciate contributions

File: src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy [hunk 1]
```diff
@@ -159,14 +159,15 @@ class ConanHostedRecipe
     createRoute(builder, uploadConaninfo(), CONAN_INFO, hostedHandler.uploadConanInfo)
     createRoute(builder, uploadConanPackageZip(), CONAN_PACKAGE, hostedHandler.uploadConanPackage)
     createRoute(builder, uploadConanSources(), AssetKind.CONAN_SOURCES, hostedHandler.uploadConanSources)
-    createRoute(builder, uploadConanExportZip(), CONAN_PACKAGE, hostedHandler.uploadConanExport)
+    createRoute(builder, uploadConanExportZip(), AssetKind.CONAN_EXPORT, hostedHandler.uploadConanExport)
```
Comment (DarthHater) [inline src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]:
```suggestion
    createRoute(builder, uploadConanExportZip(), CONAN_EXPORT, hostedHandler.uploadConanExport)
```

SUGGESTION: We should likely static import this, as well as the CONAN_SOURCES above it (don't need to do that in this PR, but it's inconsistent)

File: src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy [hunk 2]
```diff
@@ -159,14 +159,15 @@ class ConanHostedRecipe
...
 
     createRoute(builder, downloadUrls(), AssetKind.DOWNLOAD_URL, hostedHandler.downloadUrl)
     createRoute(builder, downloadManifest(), CONAN_MANIFEST, hostedHandler.download)
     createRoute(builder, downloadConanfile(), CONAN_FILE, hostedHandler.download)
     createRoute(builder, downloadConaninfo(), CONAN_INFO, hostedHandler.download)
     createRoute(builder, downloadConanPackageZip(), CONAN_PACKAGE, hostedHandler.download)
     createRoute(builder, downloadConanSources(), AssetKind.CONAN_SOURCES, hostedHandler.download)
+    createRoute(builder, downloadConanExportZip(), AssetKind.CONAN_EXPORT, hostedHandler.download)
```
Comment (DarthHater) [inline src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]:
```suggestion
    createRoute(builder, downloadConanExportZip(), CONAN_EXPORT, hostedHandler.download)
```

Same suggestion as before!

File: src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy [hunk 3]
```diff
@@ -384,6 +385,15 @@ class ConanHostedRecipe
     )
   }
 
+  static Builder downloadConanExportZip() {
```
Comment (DarthHater) [inline src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]:
```suggestion
  static Builder downloadConanExportZip(final String url) {
```

Being that this is virtually identical to downloadConanPackageZip, you could consolidate this Builder with the other by accepting a String with the URL and using that in the TokenMatcher

File: src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy [hunk 4]
```diff
@@ -384,6 +385,15 @@ class ConanHostedRecipe
...
+    new Builder().matcher(
+            and(
+                    new ActionMatcher(HEAD, GET),
+                    new TokenMatcher(CONAN_EXPORT_ZIP_URL)
```
Comment (DarthHater) [inline src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]:
```suggestion
                    new TokenMatcher(url)
```

If you take the suggestion from above

File: src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy (hunk 1 above)
Comment (remgodow) [reply src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]: CONAN_SOURCES and DOWNLOAD_URL are referenced by AssedKind due to name conflict with class member strings.

File: src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy (hunk 2 above)
Comment (remgodow) [reply src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]: Same comment as before!

**Pull request merged**